src/main/java/com/example/demo/controllers/UsersController.java

```diff
@@ -12,18 +15,23 @@ public class UsersController {
 
     private UsersService userService;
 
+    @PreAuthorize("hasAuthority('admin') or hasAuthority('user')")
     @DeleteMapping("/users")
-    public void deleteUser(@RequestHeader(value="token") String authToken) {
-        userService.deleteUser(authToken);
+    public void deleteUser() {
+        UserDetails userDetails = (UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        userService.deleteUser(userDetails.getUsername());
     }
 
+    @PreAuthorize("hasAuthority('admin')")
     @GetMapping("/users/{id}")
-    public User getUser(@PathVariable int id, @RequestHeader(value="token") String authToken) {
-        return userService.getUser(id, authToken);
+    public AppUser getUser(@PathVariable int id) {
+        return userService.getUser(id);
     }
 
+    @PreAuthorize("hasAuthority('admin') or hasAuthority('user')")
     @PutMapping("/users")
-    public User updateUser(@RequestHeader(value="token") String authToken, @RequestBody User user) {
-        return userService.updateUser(authToken, user);
+    public AppUser updateUser(@RequestBody AppUser appUser) {
+        UserDetails userDetails = (UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        return userService.updateUser(appUser, userDetails.getUsername());
```

Comment:
ok i will update this, but i made it in this way because when user sends his credentials he will provide server by his email so i generate token using his email, but i will correct this mistake